### PR TITLE
gh #5 add correct tvsettings hal specs

### PR DESCRIPTION
**Pull request by utkarshece14**
Please review

### CODE REVIEW

File: docs/pages/tv-settings_halSpec.md [hunk 1]
```diff
@@ -0,0 +1,514 @@
...
+|  |  | Vivid | Repeat as standard |
+|  |  | Graphics | Repeat as standard |
+|  | HDR10 | Repeat as SDR |
+|  | HLG | Repeat as SDR |
+|  | DV | Repeat as SDR |
+| HDMI2 | Repeat as HDMI1 |
+| HDMI3 | Repeat as HDMI1 |
+| Tunner | Repeat as HDMI1 |
+| IP | Repeat as HDMI1 |
+| Composite | Repeat as HDMI1 for SDR |
```
Comment (srinivasgtl) [inline docs/pages/tv-settings_halSpec.md:137]: Do we have different picture modes on composite?  I think we discussed this and though of setting to SDR and Entertainment picture mode.  if that is the case, can we mention the same here? same in the picture association table as well.

File: docs/pages/tv-settings_halSpec.md [hunk 2]
```diff
@@ -0,0 +1,514 @@
...
+    Note over HAL: Reload TMAX if there is a LDIM level change
+    HAL->>OPPDB: Read associated TMAX value
+    OPPDB-->>HAL:return
+    Note over HAL: Apply the new picture properties, gamma, TMAX and white balance if they have changed.
+    HAL->>Driver: Apply new picture properties
+    Driver-->>HAL:return
+    HAL->>OPPDB: Update the Picture mode association table
+    Note over HAL: Associate new picture mode to current video format and current video source
+    OPPDB-->>HAL:return
+    HAL-->>Caller:return
```
Comment (srinivasgtl) [inline docs/pages/tv-settings_halSpec.md:348]: Just reconfirming gain. Does these changes Picture parameters will get erased during the init() function call again?  how will be recover if these picture values has been modified in the previous instance (before init) and when rebooted all those stored values are gone and replaced with default ones.

File: docs/pages/tv-settings_halSpec.md [hunk 3]
```diff
@@ -0,0 +1,514 @@
...
+
+    Caller->>HAL: tvSettings_GetMethods
+    Note over HAL: APIs to get the PQ Parameters
+    HAL->>Driver:Gets the PQ Parameters
+    Driver-->>HAL:return
+    HAL-->>Caller:return
+
+    Caller->>HAL: tvSettings_SaveMethods
+    Note over HAL: APIs to save the Picture Quality Parameters
+    HAL->>OPPDB:Save the PQ Parameters
```
Comment (srinivasgtl) [inline docs/pages/tv-settings_halSpec.md:398]: Can we define which Get API will get from SoC, which ones will get from OPPDB and which one will get from DPPDB

**Review by srinivasgtl — COMMENTED** [review]
A few comments to be addressed to have a bit more clarity on our document.  Please do address these ones.

File: docs/pages/tv-settings_halSpec.md [hunk 4]
```diff
@@ -0,0 +1,514 @@
...
+style x fill:#9f9,stroke:#333,stroke-width:0.3px
+```
+	
+## Component Runtime Execution Requirements
+
+### Initialization and Startup
+
+The caller must initialize the `APIs` with picture quality modes for specific platforms and initiates communication with picture quality drivers. The standard/default values shall be maintained in default Picture profile database and any modified values for these parameters using TV Settings HAL APIs will be maintained in override Picture profile database. HAL shall be responsible to store these Picture profile data into the database.
+Picture profile database will have 5 types of tables:
+1. Picture property table to maintain all picture properties for a given picture mode, video format and video source
```
[the remark this answers is not in the document]
Comment (utkarshece14) [reply docs/pages/tv-settings_halSpec.md:77]: Done

File: docs/pages/tv-settings_halSpec.md [hunk 5]
```diff
@@ -0,0 +1,514 @@
...
+2. Picture mode association table maintain the association of a given video source and video format to a picture mode
+3. White balance table for every color temperature and video source to maintain WB calibrated values
+4. Gamma table for every color temperature to maintain the gamma calibrated values.
+5. TMAX table for every local dimming level to maintain the TMAX value
+
+- The specification of the TV picure configuration will be defined in a config file ([pq_capablities.ini](../../config/pq_capablities.ini "pq_capablities.ini")) which decides supported formats, picture modes, dimming modes, dvModes, resolution etc.
```
[the remark this answers is not in the document]
Comment (utkarshece14) [reply docs/pages/tv-settings_halSpec.md]: Done

File: docs/pages/tv-settings_halSpec.md [hunk 6]
```diff
@@ -0,0 +1,514 @@
...
+- Caller must initialize by calling `tvInit()` which must initialize the parameters in picture property table. These parameters are decided by Soc vendor  based on platform capability.
```
[the remark this answers is not in the document]
Comment (utkarshece14) [reply docs/pages/tv-settings_halSpec.md]: Done

File: docs/pages/tv-settings_halSpec.md [hunk 7]
```diff
@@ -0,0 +1,514 @@
...
+- On every bootup the default picture profile database will be a copied of override picture profile database.
```
[the remark this answers is not in the document]
Comment (utkarshece14) [reply docs/pages/tv-settings_halSpec.md]: Done

File: docs/pages/tv-settings_halSpec.md (hunk 1 above)
Comment (utkarshece14) [reply docs/pages/tv-settings_halSpec.md]: Done

File: docs/pages/tv-settings_halSpec.md (hunk 2 above)
Comment (utkarshece14) [reply docs/pages/tv-settings_halSpec.md:348]: its plugin responsibility to update with user changed values after TV init

File: docs/pages/tv-settings_halSpec.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (srinivasgtl) [reply docs/pages/tv-settings_halSpec.md:77]: The new sentence is bit misleading.  Can you add the statement as point 3 in the points stating as 
*If input source, picture mode, and video format are not already set or specified, they will default to "IP source," "Entertainment," and "SDR," respectively. However, if the caller attempts to change parameters that rely on these values, they will be acted upon default values accordingly.*

File: docs/pages/tv-settings_halSpec.md (hunk 2 above)
Comment (srinivasgtl) [reply docs/pages/tv-settings_halSpec.md:348]: So plugin will maintain the same data base with respect to the picture parameters again similar to the kind of tables we have here?

File: docs/pages/tv-settings_halSpec.md (hunk 3 above)
Comment (srinivasgtl) [reply docs/pages/tv-settings_halSpec.md:398]: This one not addressed?

File: docs/pages/tv-settings_halSpec.md (hunk 2 above)
Comment (srinivasgtl) [reply docs/pages/tv-settings_halSpec.md:348]: checked the architecture.. looks TR181 wll maintain these params.

**Review by srinivasgtl — APPROVED** [review]
all comments now fixed and resolved.

Ship it.

**Pull request closed without merging**